src/main.cpp

```diff
@@ -103,45 +103,75 @@ parse_certs(boost::json::object const& config)
 void
 initLogging(boost::json::object const& config)
 {
+    namespace src = boost::log::sources;
+    namespace keywords = boost::log::keywords;
+    namespace sinks = boost::log::sinks;
+    namespace trivial = boost::log::trivial;
     boost::log::add_common_attributes();
     std::string format = "[%TimeStamp%] [%ThreadID%] [%Severity%] %Message%";
-    boost::log::add_console_log(
-        std::cout, boost::log::keywords::format = format);
-    if (config.contains("log_file"))
+    if (config.contains("log_to_console") &&
+        config.at("log_to_console") == true)
     {
-        boost::log::add_file_log(
-            config.at("log_file").as_string().c_str(),
-            boost::log::keywords::format = format,
-            boost::log::keywords::open_mode = std::ios_base::app);
+        boost::log::add_console_log(std::cout, keywords::format = format);
+    }
+    if (config.contains("log_to_file") && config.at("log_to_file") == true &&
+        config.contains("log_directory"))
+    {
+        boost::filesystem::path dirPath{
+            config.at("log_directory").as_string().c_str()};
+        if (!boost::filesystem::exists(dirPath))
+            boost::filesystem::create_directories(dirPath);
+        const uint64_t rotationSize = config.contains("log_rotation_size")
+            ? config.at("log_rotation_size").as_uint64()
+            : 2 * 1024 * 1024 * 1024u;
+        const uint64_t rotationPeriod =
+            config.contains("log_rotation_hour_interval")
+            ? config.at("log_rotation_hour_interval").as_uint64()
+            : 12u;
+        const uint64_t dirSize = config.contains("log_directory_max_size")
+            ? config.at("log_directory_max_size").as_uint64()
+            : 50 * 1024 * 1024 * 1024u;
+        auto fileSink = boost::log::add_file_log(
+            keywords::file_name = dirPath / "clio.log",
+            keywords::target_file_name = dirPath / "clio_%Y-%m-%d_%H-%M-%S.log",
+            keywords::auto_flush = true,
+            keywords::format = format,
+            keywords::open_mode = std::ios_base::app,
+            keywords::rotation_size = rotationSize,
+            keywords::time_based_rotation =
+                sinks::file::rotation_at_time_interval(
+                    boost::posix_time::hours(rotationPeriod)));
+        fileSink->locked_backend()->set_file_collector(
+            sinks::file::make_collector(
+                keywords::target = dirPath, keywords::max_size = dirSize));
+        fileSink->locked_backend()->scan_for_files();
```

Comment:
So, does rotation happen when the `file > dirSize` or every `rotationPeriod` seconds, or both?

I.E. if we hit our limit, does that cancel out the time based rotation, or will we see something like this:
debug001.log -> 2GB
debug002.log -> 2GB
debug003.log -> 1GB // here we hit the time based rotation when the file is only 1GB
debug004.log -> 2GB

Reply:
The rotation happens whenever either of the two conditions is met (when the main log file exceeds the limit or enough time passes since the last rotation). So yes, when the log is rotated because of file size limit, the time will reset.